packages/clerk-js/src/core/clerk.ts

```diff
@@ -200,9 +201,9 @@ export class Clerk implements ClerkInterface {
   //@ts-expect-error with being undefined even though it's not possible - related to issue with ts and error thrower
   #fapiClient: FapiClient;
   #instanceType?: InstanceType;
-  #loaded = false;
-
+  #status: ClerkInterface['status'] = 'loading';
   #listeners: Array<(emission: Resources) => void> = [];
+  #statusListeners: Array<(status: Clerk['status']) => void> = [];
```

Comment:
I think we need a more general event pattern in the core Clerk class. Adding on additional properties with corresponding getters/setters to the core class is not sustainable in the long term